src/App.tsx

```diff
@@ -1,61 +1,138 @@
+import React, { useState } from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function findUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [todos, setTodos] = useState(preparedTodos);
+  const [isUserError, setUserError] = useState(false);
+  const [isTitleEror, setTitleError] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleError(!title.trim());
+    setUserError(!selectedUser);
+
+    if (title.trim() === '' || !selectedUser) {
+      return;
+    }
+
+    const addedUser = findUserById(+selectedUser);
+
+    setTodos(current => {
+      const nextTodo = Math.max(...current.map(todo => todo.id)) + 1;
+
+      return [
+        ...current,
+        {
+          id: nextTodo,
+          title,
+          completed: false,
+          userId: addedUser ? addedUser.id : null,
+          user: addedUser,
+        },
+      ];
+    });
+
+    setTitle('');
+    setSelectedUser(0);
+  };
+
+  const handleTitleChange = (
+    { target }: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTitle(target.value);
+    setTitleError(false);
```

Comment:
Why do we need to set this variable to`false` all the time? Maybe, let's do it when `isTitleEror` is not empty?